src/pods/test-multiple-choice/test-multiple-choice.business.ts

```diff
@@ -0,0 +1,36 @@
+import { VerbEntityGlobal } from 'core/verbs';
+import { VerbQuiz } from './test-multiple-choice.vm';
+import { Verb, VerbCorrect, createDefaultVerbCorrect } from 'common/model';
+
+// TODO: maybe add some defensive programming here? edge cases / errors ?
+export const pickRandomVerb = (
+  selectedVerbs: string[],
+  verbs: VerbEntityGlobal[]
+): Verb => {
+  const allVerbs = selectedVerbs.length === 0;
+  const arrayLength = allVerbs ? verbs.length : selectedVerbs.length;
+  const index = Math.floor(Math.random() * arrayLength);
+
+  const verb = allVerbs
+    ? verbs[index]
+    : verbs.find(verb => verb.infinitive === selectedVerbs[index]);
+
+  // TODO: we should create a separate mapper for this
+  return {
+    ...verb,
+  };
+};
+
+export const answerIsCorrect = (verb: Verb, quiz: VerbQuiz): VerbCorrect => {
+  let verbCorrect = createDefaultVerbCorrect();
+  verbCorrect.all = verb.infinitive === quiz.response;
+  return verbCorrect;
+};
+
+export const mixOptions = (options: Verb[]) => {
+  for (let i = options.length - 1; i > 0; i--) {
+    let j = Math.floor(Math.random() * (i + 1));
+    [options[i], options[j]] = [options[j], options[i]];
+  }
```

Comment:
This is a little bit criptic, what do we want to achieve here? My guess

- I start from the last option to the first one.
- I just pick a random number in between the current index plus one and cero.
-I swap values (mutating the array).

Before refactoring we should create some unit tests for this cases:
  - We cannot test that numbers are swapped because it could happen that it doesn't change.
  - We can try with arrays of several length.
  - We can test that array length is the same.
  - We can test that every verb on the original array are in the resulting one.

Trying to apply some clean code, first let's go for a mutation version (not ideal), then let's try an inmmutable version:

```ts
export const mixOptions = (options: Verb[]) => {
  const mixedOptions = [...options];

  for (let currentPosition = mixedOptions.length - 1; currentPosition > 0; currentPosition--) {
    let newPosition = Math.floor(Math.random() * (currentPosition + 1));
    [options[currentPosition], options[newPosition]] = [options[newPosition], options[currentPosition]];
  }

  return mixedOptions;
}
```

Reply:
I've installed and used **lodash.shuffle**